esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -3677,3 +3680,19 @@ bool CFileSprayEx::onGetDFUServerQueues(IEspContext &context, IEspGetDFUServerQu
 
     return true;
 }
+
+void CFileSprayValidateHelper::validateGetDFUWorkunitsRequest(IEspGetDFUWorkunits& req)
+{
+    if (!isEmptyString(req.getWuid()) && !looksLikeAWuid(req.getWuid(), 'D'))
+        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "GetDFUWorkunits: Invalid Workunit ID: %s", req.getWuid());
+    if (!isEmptyString(req.getOwner()) && !isValidXPathValue(req.getOwner()))
+        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "GetDFUWorkunits: Invalid Owner %s.", req.getOwner());
+    if (!isEmptyString(req.getCluster()) && !isValidXPathValue(req.getCluster()))
+        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "GetDFUWorkunits: Invalid Cluster %s.", req.getCluster());
+    if (!isEmptyString(req.getStateReq()) && !isValidXPathValue(req.getStateReq()))
+        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "GetDFUWorkunits: Invalid StateReq %s.", req.getStateReq());
+    if (!isEmptyString(req.getJobname()) && !isValidXPathValue(req.getJobname()))
+        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "GetDFUWorkunits: Invalid Jobname %s.", req.getJobname());
+    if (!isEmptyString(req.getPublisherWuid()) && !isValidXPathValue(req.getPublisherWuid()))
```

Comment:
I think for PublisherWuid you can do  !looksLikeAWuid(req.getPublisherWuid(), 'P').

Reply:
I just made the change.